test/ILLink.RoslynAnalyzer.Tests/LinkerTestCases.cs

```diff
@@ -13,20 +16,27 @@ namespace ILLink.RoslynAnalyzer.Tests
 	/// </summary>
 	public class LinkerTestCases : TestCaseUtils
 	{
-		[Theory]
-		[MemberData (nameof (TestCaseUtils.GetTestData), parameters: nameof (RequiresCapability))]
-		public void RequiresCapability (MethodDeclarationSyntax m, List<AttributeSyntax> attrs)
+		private static Lazy<IEnumerable<MetadataReference>> _additionalReferences = new Lazy<IEnumerable<MetadataReference>> (LoadDependencies);
+
+		private static IEnumerable<MetadataReference> LoadDependencies ()
```

Comment:
Why not put this in LinkerTestCases.cs? It feels like `GetReferenceFilesByDirName` should only be used there and have the dependencies resolved before compilation (ideally through looking at the linker attributes that we use to add assemblies to the compilation).